src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,131 @@
+            }
+            newElementsData[index] = value;
+            for (int i = index; i < size(); i++) {
+                newElementsData[i + 1] = elementsData[i];
+            }
+            size++;
+            elementsData = newElementsData;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Index doesn't found");
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < size && index >= 0) {
+            return elementsData[index];
+        }
+        throw new ArrayListIndexOutOfBoundsException("Index out of bounds for index: " + index);
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index < size && index >= 0) {
+            elementsData[index] = value;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Index doesn't found");
+        }
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        T remove;
+        if (index < size && index >= 0) {
+            T[] newElementsData = (T[]) new Object[capacity];
+            for (int i = 0; i < index; i++) {
+                newElementsData[i] = elementsData[i];
+            }
+            remove = elementsData[index];
+            for (int i = index; i < size(); i++) {
+                newElementsData[i] = elementsData[i + 1];
+            }
+            size--;
+            elementsData = newElementsData;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Index doesn't found");
+        }
+        return remove;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        int ourIndex = getIndexOfElement(element);
+        if (element == null && elementsData[ourIndex] == null
+                || element != null && elementsData[ourIndex].equals(element)) {
+            return remove(ourIndex);
+        } else {
+            throw new NoSuchElementException("No such element: " + element);
+        }
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void resizeArray() {
+        double newCapacity = capacity * DEFAULT_MULTIPLICATION_NUMBER;
+        this.capacity = (int) newCapacity;
+        T[] newElementsData = (T[]) new Object[capacity];
+        System.arraycopy(elementsData, 0, newElementsData, 0, size);
+        elementsData = newElementsData;
+    }
+
+    private int getIndexOfElement(T element) {
+        for (int i = 0; i < size(); i++) {
+            if (elementsData[i] == element || elementsData[i] != null
+                    && elementsData[i].equals(element)) {
+                return i;
+            }
+        }
+        return 0;
```

Comment:
The method getIndexOfElement(T element) should throw an exception if the element is not found instead of returning 0. Returning 0 could be misleading because it implies that the element is found at index 0.